presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -365,6 +366,12 @@ private void runCommit(long transactionId, HandleConsumer callback)
                 return;
             }
             catch (DBIException e) {
+                if (isTransactionCacheFullError(e)) {
+                    throw metadataError(e,
+                            "A possible cause of this problem is the transaction size is too big " +
```

Comment:
"Transaction size is too big" is a sure cause, because that's the MySQL error you're getting. However, the reasons for getting it are hypothetical. It would be good to adjust the message for it.
Also, "inserting rows" in a user facing message might suggest that actual number of rows is high, but what it should really say is that the number of shards (=rows in mysql table) is high.
@electrum please suggest message improvements.